### PR TITLE
Visual Label better than aria-label

### PR DESCRIPTION
**Pull request by shweta786**
PR to suggest that Visual label is better than aria-label

https://github.com/microsoft/eslint-plugin-fluentui-jsx-a11y/issues/59

### CODE REVIEW

File: lib/rules/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+module.exports = {
```
Comment (aubreyquinn) [inline lib/rules/index.js:4]: Uh oh, is this a merge conflict?

File: lib/rules/visual-label-better-than-aria-suggestion-v9.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+"use strict";
+
+var elementType = require("jsx-ast-utils").elementType;
+const { hasNonEmptyProp } = require("../util/hasNonEmptyProp");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+// Define an array of applicable component names
+const allowedComponents = ["Dropdown", "SpinButton", "CompoundButton"];
```
Comment (aubreyquinn) [inline lib/rules/visual-label-better-than-aria-suggestion-v9.js:13]: Can we move this out into the applicableComponents folder? Also, Inputs and other types of buttons apply to this rule.

File: lib/rules/index.js (hunk 1 above)
Comment (shweta786) [reply lib/rules/index.js]: yupp, Got little confusion but resolved now

File: tests/lib/rules/visual-label-better-than-aria-suggestion-v9.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+ruleTester.run("visual-label-better-than-aria-suggestion-v9", rule, {
+    valid: [
+        `<><Label id="my-dropdownid">This is the visual label</Label><Dropdown aria-labelledby="my-dropdownid" /></>`,
```
Comment (aubreyquinn) [inline tests/lib/rules/visual-label-better-than-aria-suggestion-v9.js:20]: I think we can pass in the applicable components to generate the test cases like in this file: input-components-require-accessible-nam

File: lib/index.ts [hunk 4]
```diff
@@ -40,7 +40,8 @@ module.exports = {
         "dialogsurface-needs-aria": require("./rules/dialogsurface-needs-aria"),
         "spinner-needs-labelling": require("./rules/spinner-needs-labelling"),
         "badge-needs-accessible-name": require("./rules/badge-needs-accessible-name"),
-        "progressbar-needs-labelling": require("./rules/progressbar-needs-labelling")
+        "progressbar-needs-labelling": require("./rules/progressbar-needs-labelling"),
+        "visual-label-better-than-aria-suggestion-v9": require("./rules/visual-label-better-than-aria-suggestion-v9")
```
Comment (aubreyquinn) [inline lib/index.ts:44]: We also need to add this one to the config object below :)

File: docs/rules/visual-label-better-than-aria-suggestion-v9.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Accessibility: Visual labels should be preferred over aria-label (`@microsoft/fluentui-jsx-a11y/visual-label-better-than-aria-suggestion-v9`)
```
Comment (aubreyquinn) [inline docs/rules/visual-label-better-than-aria-suggestion-v9.md:1]: We don't need -v9 on the end anymore, I got rid of the v8 rules ;)

Comment (aubreyquinn) [inline docs/rules/visual-label-better-than-aria-suggestion-v9.md:1]: I think there are still linting issues on the readme file :)

**Review by aubreyquinn — CHANGES_REQUESTED** [review]
comment :)

File: docs/rules/visual-label-better-than-aria-suggestion-v9.md (hunk 5 above)
Comment (shweta786) [reply docs/rules/visual-label-better-than-aria-suggestion-v9.md]: I tried running the md generator command but it's changing some random files and my file was passing lint when trying locally. @aubreyquinn can you please indicate the issue if you can figure out in this readme file

**Pull request merged**